### ISSUE TITLE
retrieve id token (or reauth) when sending password-protected message

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#1222]
part of #955, after #1091 and #1221

When sending password-protected message as described in #1221, we'll soon be uploading the message to FES where it will be served through a web portal. Similar to https://flowcrypt.com/docs/guide/send-and-receive/send-password-protected.html

For this, we'll need an active / not expired id token.

Before sending, please check, using #1221, whether we have an id token that still has at least 5 minutes left before expiring (to account for encrypting / sending etc).

If not (don't have any or it's expired), please show `Please log in with Google to continue` and then show Google prompt to authenticate (only with id token scope). That will refresh the id token. Once it's fresh, the sending process may continue.

The id token won't actually be used in this issue yet, that will be for another issue.

You can test this in browser extension, which already does this (when sending from flowcrypt.com account).
